File: APTJava/apt-processor/src/main/java/com/example/apt_compiler/launcher/AutoBind.java
package com.example.apt_compiler.launcher;


import com.example.apt_compiler.template.IBindHelper;

public class AutoBind {
    private static volatile AutoBind instance = null;

    public static AutoBind getInstance() {
        if(instance == null) {
            synchronized (AutoBind.class) {
                if (instance == null) {
                    instance = new AutoBind();
                }
            }
        }
        return instance;
    }

    public void inject(Object target) {
        String className = target.getClass().getCanonicalName();
        String helperName = className + "$$Autobind";
        try {
            IBindHelper helper = (IBindHelper) (Class.forName(helperName).getConstructor().newInstance());

            helper.inject(target);
        }   catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: APTKotlin/apt-processor/src/main/java/com/example/apt_compiler/launcher/AutoBind.kt
package com.example.apt_compiler.launcher

import com.example.apt_compiler.template.IBindHelper

object AutoBind {
    fun inject(target: Any) {
        val className = target.javaClass.canonicalName
        val helperName = "$className$\$Processor"
        try {
            val helper = Class.forName(helperName).getConstructor().newInstance() as IBindHelper
            helper.inject(target)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}